src/Service/InstallationService.php

```diff
@@ -939,6 +943,77 @@ private function addSchemasToCollection(CollectionEntity $collection, string $sc
 
     }//end addSchemasToCollection()
 
+    /**
+     * This function creates templates for an array of schema references.
+     *
+     * @param array $templatesData An array of data used for creating templates.
+     *
+     * @return array An array of templates
+     */
+    private function createTemplates(array $templatesData = []): array
+    {
+        $templates = [];
+
+        // Let's loop through the templatesData.
+        foreach ($templatesData as $templateData) {
+            // Create the base Endpoint.
+            $template = $this->createTemplate($templateData);
+            if ($template === null) {
+                continue;
+            }
+
+            $templates[] = $template;
+        }//end foreach
+
+        $this->logger->info(count($templates).' Templates Created');
+
+        return $templates;
+
+    }//end createTemplates()
+
+    /**
+     * Creates a single endpoint for an Entity or a Source using the data from installation.json.
+     *
+     * @param array $templateData The data used to create an Template.
+     *
+     * @return Template|null The created Template or null.
+     */
+    private function createTemplate(array $templateData): ?Template
+    {
+        $repository = $this->entityManager->getRepository('App:Entity');
+
+        $supportedSchemas = [];
+        foreach ($templateData['supportedSchemas'] as $schema) {
+            $supportedSchemas[] = $this->checkIfObjectExists($repository, $schema, 'Template')->getId()->toString();
+        }
+
+        unset($templateData['supportedSchemas']);
+        $templateData['supportedSchemas'] = $supportedSchemas;
+
+        if (key_exists('organization', $templateData) === true) {
+            $orgRepository = $this->entityManager->getRepository('App:Organization');
+            $organization  = $this->checkIfObjectExists($orgRepository, $templateData['organization'], 'Organization');
+
+            unset($templateData['organization']);
+            $templateData['organization'] = $organization;
+        }
+
+        $template = $this->entityManager->getRepository('App:Template')->findOneBy(['reference' => $templateData['$id']]);
+        if ($template !== null) {
+            $this->logger->debug('Template found with reference '.$templateData['$id']);
+
+            return null;
```

Comment:
Is it a conscious decision that null is returned here instead of maybe the $template ?